js/data/data.js

```diff
@@ -0,0 +1,124 @@
+var Firebase = require('firebase');
+var FirebaseUtils = require('../utils/firebase');
+
+// TODO put the uri in a constant / env var
+var FirebaseRef = new Firebase('https://quack.firebaseio.com');
+
+module.exports = {
+  create: function(resourceType, options) {
+    var handlers = {
+      message: function() {
+        if (!options || !options.message) {
+          console.log('No message specified.');
+          if (options && options.error) {
+            options.error();
+          }
+          return;
+        }
+
+        // hard-code "bestcohort" for now
+        var messagesRef = FirebaseRef.child('messages/bestcohort');
+        messagesRef.push(
+          options.message,
+          function(err) {
+            if (err) {
+              if (options && options.error) {
+                options.error(err);
+              }
+              return;
+            }
+
+            if (options && options.success) {
+              options.success();
+            }
+          }
+        );
+      }
+    };
+
+    if (handlers[resourceType]) {
+      handlers[resourceType]();
+    }
+  },
+
+  get: function(resourceType, options) {
+    var handlers = {
+      messages: function() {
```

Comment:
Actually, it looks like it shouldn't go in any handler, but before the handler is called.